src/S3/StreamWrapper.php

```diff
@@ -515,6 +515,7 @@ private function formatKey($key)
      */
     public function rename($path_from, $path_to)
     {
+        $this->initProtocol($path_from);
```

Comment:
This should be safe because PHP natively ensures a rename is issued on the same wrapper.